toolbox/githubctl/main.go

```diff
@@ -45,14 +60,111 @@ func fastForward(repo, baseBranch, refSHA *string) error {
 	return githubClnt.FastForward(*repo, *baseBranch, *refSHA)
 }
 
+func createIstioReleaseTag() error {
+	assertNotEmpty("base_branch", baseBranch)
+	pickledDeps, err := githubClnt.GetFileContent(istioRepo, *baseBranch, istioDepsFile)
+	if err != nil {
+		return err
+	}
+	deps, err := u.DeserializeDepsFromString(pickledDeps)
+	if err != nil {
+		return err
+	}
+	releaseTag, err := githubClnt.GetFileContent(istioRepo, *baseBranch, releaseTagFile)
+	if err != nil {
+		return err
+	}
+	releaseTag = strings.Trim(releaseTag, "\n")
+	releaseMsg := "Istio Release " + releaseTag
+	for _, dep := range deps {
+		if err := githubClnt.CreateAnnotatedTag(
+			dep.RepoName, releaseTag, dep.LastStableSHA, releaseMsg); err != nil {
```

Comment:
This is a valid concern. There is a chance that someone may run updateVersion.sh directly without also updating istio.deps. In latest commit, I read the sha from istio.VERSION to make sure we get the right thing.